### PR TITLE
adding pi60

### PR DESCRIPTION
**Pull request by ziptyze**
Adding pi60 to 1upkeyboards

## Description

Adding the pi60, a multi layout solderable 60% pcb

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/1upkeyboards/pi60/config.h [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "config_common.h"
+
+#define DYNAMIC_KEYMAP_LAYER_COUNT 10
+
+#define NOP_FUDGE 0.4
+
```
Comment (waffle87) [inline keyboards/1upkeyboards/pi60/config.h:11]:
```suggestion
```

File: keyboards/1upkeyboards/pi60/info.json [hunk 2]
```diff
@@ -0,0 +1,150 @@
+{
+    "manufacturer": "1upkeyboards",
+    "keyboard_name": "pi60",
+    "maintainer": "ziptyze",
+    "processor": "RP2040",
+    "bootloader": "rp2040",
+    "board": "GENERIC_RP_RP2040",
+    "usb": {
+        "device_ver": "0x0001",
```
Comment (waffle87) [inline keyboards/1upkeyboards/pi60/info.json:9]:
```suggestion
```

File: keyboards/1upkeyboards/pi60/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,45 @@
+#include QMK_KEYBOARD_H
```
Comment (waffle87) [inline keyboards/1upkeyboards/pi60/keymaps/default/keymap.c:1]: This file is missing a license header.

File: keyboards/1upkeyboards/pi60/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,99 @@
+#include QMK_KEYBOARD_H
```
Comment (waffle87) [inline keyboards/1upkeyboards/pi60/keymaps/via/keymap.c:1]: This file is missing a license header.

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/1upkeyboards/pi60/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+    [2] = LAYOUT_60_ansi(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        RGB_TOG, RGB_MOD, RGB_HUI, RGB_SAI, RGB_VAI, RGB_SPI, RGB_M_P, RGB_M_B, RGB_M_R, RGB_M_SW,KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, RGB_RMOD,RGB_HUD, RGB_SAD, RGB_VAD, RGB_SPD, RGB_M_SN,RGB_M_K, RGB_M_X, RGB_M_G, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS,                KC_TRNS, KC_TRNS, KC_TRNS,             KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+    [3] = LAYOUT_60_ansi(
```
Comment (drashna) [inline keyboards/1upkeyboards/pi60/keymaps/via/keymap.c:45]: Just a heads up, VIA should fill blank layers correctly on init, so you shouldn't need to fill out all of the layers.

**Review by drashna — APPROVED** [review]

**Review by keyboard-magpie — APPROVED** [review]

**Pull request merged**